mullvad-relay-selector/src/relay_selector/matcher.rs

```diff
@@ -0,0 +1,282 @@
+//! This module is responsible for filtering the whole relay list based on queries.
+use mullvad_types::{
+    constraints::{Constraint, Match},
+    custom_list::CustomListsSettings,
+    relay_constraints::{
+        BridgeState, InternalBridgeConstraints, Ownership, Providers, ResolvedLocationConstraint,
+        TransportPort,
+    },
+    relay_list::{
+        OpenVpnEndpoint, OpenVpnEndpointData, Relay, RelayEndpointData, WireguardEndpointData,
+    },
+};
+use talpid_types::net::{IpVersion, TransportProtocol, TunnelType};
+
+use super::query::{OpenVpnRelayQuery, RelayQuery, WireguardRelayQuery};
+
+#[derive(Clone)]
+pub struct RelayMatcher<T: EndpointMatcher> {
```

Comment:
I can see that you added a lot of comments to this file after moving it, which is great, but he `RelayMatcher` may deserve a docstring too